clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -409,6 +409,20 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                 instr_free(GLOBAL_DCONTEXT, &instr);
             }
         }
+        bool saw_double_syscall_instrs_with_same_pc = false;
+        if (shard->prev_instr_.instr.addr != 0 /*first*/) {
+            instr_t curr_instr;
+            instr_init(GLOBAL_DCONTEXT, &curr_instr);
+            decode_from_copy(GLOBAL_DCONTEXT, const_cast<app_pc>(memref.instr.encoding),
```

Comment:
Re the Windows crash: it should help to check the return value of this call here.